src/SystemSolar/First.cpp

```diff
@@ -1,74 +1,185 @@
+        return {x * k, y * k};
+    }
+    constexpr Vector operator/(double k) const
     {
-        return {x - other.x, y - other.y};
+        return {x / k, y / k};
     }
 
-    Vector<T> operator*(T other) const 
+    friend constexpr Vector operator*(double k, Vector const& v)
     {
-        return {x * other, y * other};
+        return v * k;
     }
+};
 
-     Vector<T> operator/(T other) const 
+inline double magnitude(Vector<double> const& v)
+{
+    return std::sqrt(v.x * v.x + v.y * v.y);
+}
+inline Vector<double> normalize(Vector<double> const& v)
+{
+    double m = magnitude(v);
+    if (m == 0.0)
     {
-        return {x / other, y / other};
+        return {0.0, 0.0};
     }
+    else
+    {
+        return v / m;
+    }
+}
 
+struct MetersTag
+{
+};
+struct SecondsTag
+{
+};
+struct KilogramsTag
+{
+};
+struct NewtonsTag
+{
+};
+struct MetersPerSecondTag
+{
 };
 
-using Vec = Vector<double>;
-    
-struct Body 
+using Meters = StrongType<double, MetersTag>;
+using Seconds = StrongType<double, SecondsTag>;
+using Kilograms = StrongType<double, KilogramsTag>;
+using Newtons = StrongType<double, NewtonsTag>;
+using MetersPerSecond = StrongType<double, MetersPerSecondTag>;
+
+using PositionVec = Vector<Meters>;
+using VelocityVec = Vector<MetersPerSecond>;
+using ForceVec = Vector<Newtons>;
+using Vec = ForceVec;
+
+struct Body
 {
     std::string name;
-    double mass;       
-    Vec position;  
-    Vec velocity;  
+    Kilograms mass;
+    PositionVec position;
+    VelocityVec velocity;
 };
 
-// Constante gravitacional
-const double G = 6.67430e-11;
+// Constant gravitational
+constexpr double G = 6.67430e-11;
 
-Vec gravityForce(const Body& a, const Body& b)
+Vec gravityForce(Body const& a, Body const& b)
 {
-    //vector diference a to b
-    Vec diff= b.position - a.position;
-
-    //distance
-    double dist = std::sqrt(diff.x * diff.x + diff.y * diff.y);
-
-    // TODO: calcular distancia y normalizar vector
-    Vec unit = diff * (1.0 / dist);
-    // TODO: aplicar fórmula de Newton: F = G * m1 * m2 / r^2
-    double forceMag= G*((a.mass * b.mass)) / (dist * dist);
-    // TODO: devolver fuerza como vector
-    return unit * forceMag;
+    double const aux = 0.0;
+    double const dx = b.position.x.get() - a.position.x.get();
+    double const dy = b.position.y.get() - a.position.y.get();
+
+    Vector<double> const diff{dx, dy};
+    double const dist2 = (diff.x * diff.x) + (diff.y * diff.y);
+    if (dist2 == aux)
+    {
+        return {aux, aux}; 
+    }
+
+    double const dist = std::sqrt(dist2);
+    Vector<double> const unit = diff / dist; 
+    Newtons const forceMag{G * (a.mass.get() * b.mass.get()) / dist2};
+
+    return {forceMag * unit.x, forceMag * unit.y};
 }
 
-int main() 
+int main()
 {
-    Body sun{"Sol", 1.99e30, {0, 0}, {0, 0}};
-    Body earth{"Tierra", 5.97e24, {1.5e11, 0}, {0, 29780}}; // 29.78 km/s
+    Body sun{"Sol", Kilograms{1.99e30}, PositionVec{0.0, 0.0}, VelocityVec{0.0, 0.0}};
```

Comment:
#should use English names and display output in English too. If you want it in Spanish, implement multi-language support and have both English and Spanish!